app/src/main/java/com/android/solvit/shared/model/chat/ChatAssistantViewModel.kt

```diff
@@ -147,16 +189,85 @@ class ChatAssistantViewModel() : ViewModel() {
    * @param onResponse Callback to handle the response
    */
   fun generateMessage(input: String, onResponse: (String) -> Unit) {
-    val prompt = buildPrompt(input)
+    val prompt = buildMessagePrompt(input)
     Log.d("ChatAssistantViewModel", "model prompted")
     viewModelScope.launch {
       try {
-        val response = model.generateContent(prompt)
+        val response = messagesModel.generateContent(prompt)
         response.text?.let { onResponse(it) }
       } catch (e: Exception) {
         Log.e("ChatAssistantViewModel", "Error generating response", e)
         onResponse("Sorry, I couldn't process that.")
       }
     }
   }
+
+  /**
+   * Build the prompt for the suggestions assistant
+   *
+   * @return The generated prompt
+   */
+  fun buildSuggestionsPrompt(): String {
+    var prompt = "Provide a list without repetitions of suggestions themes for a response"
+    if (_messageContext.value.isNotEmpty()) {
+      prompt += ", based on the following conversation:\n"
+      prompt += messageContext.value.joinToString("\n") { it.senderName + ": " + it.message }
```

Comment:
Method will have to be updated because we use know also image messages but it's normal since the PR handling different format types is not yet merged